tools/unstableWarningAnonymizer/unstableAnonScript.chpl

```diff
@@ -85,66 +89,75 @@ proc countUniqueWarnings(ref warningsMap: map(string, ?t), inputFileReader: IO.f
     } else {
       warningsMap[warning][0] = 1;
     }
-    warningsMap[warning][1].add(fileName); // Add the fileName to the set of files that have this warning
+    // Add the fileName to the set of files that have this warning
+    warningsMap[warning][1].add(fileName);
   }
 }
 
-proc containsUnstableWarning(warning: string) : bool {
+proc containsUnstableWarning(warning: string): bool {
   // Check if the string mentions that something is unstable
-  return warning.find("unstable") != -1 || warning.find("enum-to-bool") != -1 ||
-         warning.find("enum-to-float") != -1;
+  return warning.find("unstable") != -1 || warning.find("enum-to-bool") != -1
+         || warning.find("enum-to-float") != -1;
 }
 
 
-proc anonymizeWarning(warning: string) : string {
+proc anonymizeWarning(warning: string): string {
   // Anonymize known warning messages that include variable names
   // when so that it doesn't reveal variable names or other impl details
 
-  const typeName = "warning: using a type's name ";
-  const typeNameUse = "in a 'use' statement to access its tertiary methods is an unstable feature";
-  const typeNameImport = "in an 'import' statement to access its tertiary methods is an unstable feature";
+  param typeName = "warning: using a type's name ";
+  param typeNameUse = "in a 'use' statement to access its tertiary methods " +
+                      "is an unstable feature";
+  param typeNameImport = "in an 'import' statement to access its tertiary " +
```

Comment:
You could probably use the if statement to put "a 'use'"/"an 'import'" in using `format`, but this also works.

Reply:
I'm not sure if there's a concise way that `format` would allow me to be as strict with the error checking. How I first thought this could work with `format` was:
```chapel
param typeName = "warning: using a type's name ";
param typeNameUse = "in %s statement to access its tertiary methods is an unstable feature";
if warning.find(typeName) != 1 {
  if warning.find("a 'use'") != -1 then
    return typeName + typeNameUse.format("a 'use'");
  else if warning.find("an 'import'") != -1 then
    return typeName + typeNameUse.format("an 'import'");
}
```

But that is checking fewer characters in the `warning` string and therefore has a slightly higher chance of a false positive because user code could print anything.

Is there another way that could allow me to be strict but also not use too many `find` calls on the string?

Reply:
Ah, I see what you mean.  No, I think what you currently have is the right call